src/plugins/BlockingCommandPlugin.ts

```diff
@@ -146,10 +146,32 @@ export default class BlockingCommandPlugin extends AbstractPlugin {
       unblockedJids.forEach(jid => {
          if (!newList.includes(jid)) {
             getElements(jid).removeClass('jsxc-blocked');
+
+            let contact = this.pluginAPI.getContact(new JID(jid));
+            if (contact)
+            {
+                let chatwindow =  contact.getChatWindow();
+                if (chatwindow)
+                {
+                   chatwindow.getElement().removeClass('jsxc-blocked');
+                }
+            }
          }
       });
 
-      blockedJids.forEach(jid => getElements(jid).addClass('jsxc-blocked'));
+      blockedJids.forEach((jid) => {
+          getElements(jid).addClass('jsxc-blocked');
+
+        let contact = this.pluginAPI.getContact(new JID(jid));
+        if (contact)
```

Comment:
Btw this works only for full jids. To close all blocked jids we would need to go through the complete contact list and check every jid. Personally I think if a user blocks some jid, he can also close the chat window :-). Maybe the better idea would be to set the status (via `contact.setStatus()`) to blocked. This should be done in the `onBlocklistUpdate` since the `onBlocklistChanged` is executed in every tab.

Reply:
ok so should it still be done ?

Reply:
If you like. Sure.